packages/core/content-manager/server/src/services/document-metadata.ts

```diff
@@ -186,10 +234,11 @@ export default ({ strapi }: { strapi: Core.Strapi }) => ({
   ) {
     // TODO: Ignore publishedAt if availableStatus=false, and ignore locale if i18n is disabled
     // TODO: Sanitize createdBy
+
     const versions = await strapi.db.query(uid).findMany({
       where: { documentId: version.documentId },
-      select: ['createdAt', 'updatedAt', 'locale', 'publishedAt', 'documentId'],
       populate: {
+        ...getDeepPopulate(uid),
         createdBy: {
```

Comment:
(assuming this is true and I'm not wrong about how it's working) it might be worth a comment that without these manual createdBy and updatedBy, the deep populate will cause all creator fields to be present, otherwise they could end up removed and introduce a security issue. Or maybe add a test that ensures certain fields like password, reset token, etc, aren't present in the results so nobody is even able to break it.